src/Runner.Listener/JobDispatcher.cs

```diff
@@ -1077,7 +1077,24 @@ private async Task CompleteJobRequestAsync(int poolId, Pipelines.AgentJobRequest
 
             if (this._isRunServiceJob)
             {
-                Trace.Verbose($"Skip FinishAgentRequest call from Listener because MessageType is {message.MessageType}");
+                var runServer = await GetRunServerAsync(systemConnection);
+                var errorMessage = GetErrorMessageFromWorkerFailDetailInfo(message, detailInfo);
+                var unhandledExceptionIssue = new Issue() { Type = IssueType.Error, Message = errorMessage };
+                var unhandledAnnotation = unhandledExceptionIssue.ToAnnotation();
+                var jobAnnotations = new List<Annotation>();
+                if (unhandledAnnotation.HasValue)
+                {
+                    jobAnnotations.Add(unhandledAnnotation.Value);
+                }
+                try
+                {
+                    await runServer.CompleteJobAsync(message.Plan.PlanId, message.JobId, TaskResult.Failed, outputs: null, stepResults: null, jobAnnotations: jobAnnotations, CancellationToken.None);
```

Comment:
```suggestion
                    await runServer.CompleteJobAsync(message.Plan.PlanId, message.JobId, result, outputs: null, stepResults: null, jobAnnotations: jobAnnotations, CancellationToken.None);
```

Reply:
ah! good catch, thank you! copy pasta error